### PR TITLE
C04_P03 C04_P04 C04_P05 C04_P07 C05_P01 C05_P02 C08_P02 - Additional solutions

### PR DESCRIPTION
**Pull request by nof20**
Some more solutions to problems

### CODE REVIEW

**Comment by brycedrennan** [comment]
@nof20 I've included a modified version of the solution you provided for Chapter 4, problem 3 here: https://github.com/careercup/CtCI-6th-Edition-Python/commit/4b95f8842118f7d75c7c78fd5ca07a0d2cab23a6

**Comment by nof20** [comment]
Awesome!

**Comment by brycedrennan** [comment]
Chapter 4, Problem 4 solution included here: https://github.com/careercup/CtCI-6th-Edition-Python/commit/e192ea43867a639dfab4031d87edca26d108a291

(I'm working through these one problem at a time)

**Comment by brycedrennan** [comment]
Chapter 4, Problem 5: https://github.com/careercup/CtCI-6th-Edition-Python/commit/f55ab4e8110edcd699a7f08071b34423512cb8be

File: Chapter5/52BinaryToString.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+"""Convert a float to IEEE 32-bit floating point notation.
+1 sign bit, 8 exponent bits with +127 (7 bit) bias, 23 mantissa bits.
+Requires Python 3 (float division)
+"""
```
Comment (brycedrennan) [inline Chapter5/52BinaryToString.py:4]: This doesn't look like it's solving a problem in the book but perhaps I've missed something?

**Comment by brycedrennan** [comment]
Okay I've wrapped up looking over this.  Thanks for all these solutions!  Some of the other solutions have since had answers merged that we'd need to integrate with.   

Given how old this pull request is I'm closing for now but happy to reopen if you desire.

**Pull request closed without merging**